### DISCUSSION TITLE
AI Deck not working

### DISCUSSION THREAD

**Discussion by akshaysunkara** [bitcraze/discussions#1476] [Q&A]
Hey guys, I'm trying to use my AI deck and am following this guide: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example

When at the Gap8 Bootloader part, and run the following code:

docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'

I get this error:

bitcraze@bitcraze-vm:~/aideck-gap8-bootloader$ docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader  
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader  
Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/olimex-arm-usb-tiny-h.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 76832 /gap_sdk/utils/openocd_tools; exit;"
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
TARGET create
GAP8 INIT TARGET
Error: JTAG scan chain interrogation failed: all ones
Error: Check JTAG interface, timings, target power, etc.
Error: Trying to use configured scan chain anyway...
Error: gap8.cpu: IR capture error; saw 0x0f not 0x01
GAP8 examine target
Init jtag
Initialising GAP8 JTAG TAP
Error: Burst read failed
Error: Burst read failed
Error: Burst read failed
can't read halt state
Error: Couldn't read the CPU state
Error: Burst read failed
can't read halt state
Init jtag
Initialising GAP8 JTAG TAP

The AI deck seems to be connected properly, its blinking green. I also attached the JTAG, the Olimex debugger.

https://github.com/user-attachments/assets/ba364450-becc-4d4c-aa3c-0aef31b4bac9
![image](https://github.com/user-attachments/assets/f44ae15c-dec8-4098-b12f-60a7f5711de2)

Just wondering if I'm doing anything wrong and what I need to fix

**Comment by akshaysunkara** [comment]
After I closed the terminal, the LED2 led was blinking green. I tried running the same command again to the same error lines.

**Comment by knmcguire** [comment]
You need to input the olimex jtag adapter pins to the connecter of the big lint cable attached to the olimex itself. Currently the pins of the gap8 are not connected to anything at the moment.

See https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/jtag-programmer/
